deepchecks/checks/methodology/test_dataset_size.py

```diff
@@ -0,0 +1,92 @@
+import typing as t
+from deepchecks import Dataset, SingleDatasetBaseCheck, CheckResult, ConditionResult
+
+
+__all__ = ['TestDatasetSize',]
+
+
+TTestSize = t.TypeVar('TTestSize', bound='TestDatasetSize')
+
+
+class TestDatasetSize(SingleDatasetBaseCheck):
+    """Verify Test dataset size."""
+
+    def run(self, train: Dataset, test: Dataset, model: object) -> CheckResult:
+        """Run check instance.
+
+        Args:
+            train (Dataset): train dataset
+            test (Dataset): test dataset
+            model (object): a scikit-learn-compatible fitted estimator instance
+
+        Returns:
+            CheckResult: with value of type Dict[str, int].
+                Value contains two keys, 'train' - size of the train dataset
+                and 'test' - size of the test dataset.
+
+        Raises:
+            DeepchecksValueError:
+                if not dataset instances were provided;
+                if datasets are empty;
+                if datasets do not have features columns;
+                if datasets do not have label column;
+                if datasets do not share same features;
+        """
+        check_name = type(self).__name__
+        Dataset.validate_dataset(train, check_name)
+        Dataset.validate_dataset(test, check_name)
+        train.validate_features(check_name)
+        train.validate_label(check_name)
+        test.validate_features(check_name)
+        test.validate_label(check_name)
```

Comment:
no need to validate label, shared features

Reply:
updated